dap_prinz_green_jobs/pipeline/green_measures/industries/prodigy/custom_openai_recipe.py

```diff
@@ -0,0 +1,181 @@
+"""
+Custom prodigy recipe to generate both NER start and end tokens of company description
+    and the predicted SIC code using an LLM.
+
+To run the custom recipe:
+
+prodigy oa_ner_classification comp_sic_annotated \
+    dap_prinz_green_jobs/pipeline/green_measures/industries/prodigy/data/mixed_ojo_sample_5000.jsonl \
+    -F dap_prinz_green_jobs/pipeline/green_measures/industries/prodigy/custom_openai_recipe.py
+"""
+import langchain
+from langchain.llms import OpenAIChat
+from langchain.output_parsers import StructuredOutputParser, ResponseSchema
+from langchain.prompts import PromptTemplate
+from pathlib import Path
+
+from dotenv import load_dotenv
+import copy
+import os
+import re
+from typing import Iterator, List, Union
+
+import prodigy
+import spacy
+from prodigy.components.loaders import JSONL
+from prodigy.components.preprocess import add_tokens
+
+load_dotenv()  # load the openAI key
+
+if not os.getenv("OPENAI_API_KEY"):
+    ValueError(
+        "OPENAI_API_KEY not found in environment variables. Add key to .env file."
+    )
+
+model = OpenAIChat(model="gpt-3.5-turbo", temperature=0.0)
+
+ner_schema = [
+    ResponseSchema(
+        name="sic_phrase",
+        description="Predict the company description sentence in the input text. It must be an exact phrase from the input text.",
```

Comment:
what happens if there are no company description sentences? does it always try to predict something?

Reply:
seems like it does although that's a really good question, i'm not sure